### PR TITLE
members.not.loading

### PR DESCRIPTION
**Pull request by nodatall**

### CODE REVIEW

File: test/organizationMembership.matchers.spec.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const { expect } = require('./matchers');
+require('./patterns');
+
+describe('organization membership matchers', function(){
```
Comment (deadlyicon) [inline test/organizationMembership.matchers.spec.js:6]:
```suggestion
describe('organizationMembership.matchers', function(){
```

File: test/organizationMemberships.matchers.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    curator: _.isUndefinedOr(_.isBoolean),
+    updatedAt: _.isUndefinedOr(_.isISODateString),
+  }
+);
+
+definePattern(
+  'anOrganizationAdminMembership',
+  target => {
+    expect(target).to.be.anOrganizationMembership();
+    expect(target.admin).to.be.true;
```
Comment (deadlyicon) [inline test/organizationMemberships.matchers.js:22]:
```suggestion
    expect(target).to.matchPattern({ admin: true, '...': 1 });
```

yields a nicer error message

File: test/patterns.js [hunk 3]
```diff
@@ -38,3 +37,10 @@ definePattern('includedIn', (target, set) =>
 definePattern('trueOrUndefined', value =>
   value === true || value === undefined
 );
+
+definePattern('anAssetUrl', /https?:\/\/.+/);
+
+definePattern('aUid', /^[a-z0-9]{32}$/);
+
+require ('./publicProfile.matchers');
+require ('./organizationMemberships.matchers');
```
Comment (deadlyicon) [inline test/patterns.js:46]: no

File: test/patterns.js [hunk 4]
```diff
@@ -38,3 +37,10 @@ definePattern('includedIn', (target, set) =>
 definePattern('trueOrUndefined', value =>
   value === true || value === undefined
 );
+
+definePattern('anAssetUrl', /https?:\/\/.+/);
```
Comment (deadlyicon) [inline test/patterns.js:41]: this seems useless

File: test/publicProfile.matchers.spec.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const { expect } = require('./matchers');
```
Comment (deadlyicon) [inline test/publicProfile.matchers.spec.js:3]: why are you not requiring the thing you are testing?

File: test/patterns.js (hunk 3 above)
Comment (deadlyicon) [reply test/patterns.js:46]: we are not putting every single matcher in one giant file.

File: test/publicProfile.matchers.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    avatar: _.isAssetUrl,
+    banner: _.isUndefinedOr(_.isAssetUrl),
+    bio: _.isUndefinedOr(_.isString),
+    city: _.isUndefinedOr(_.isString),
+    country: _.isUndefinedOr(_.isString),
+    did: _.isDID,
+    displayName: _.isString,
+    postalCode: _.isUndefinedOr(_.isString),
+    province: _.isUndefinedOr(_.isString),
+    username: _.isString,
```
Comment (deadlyicon) [inline test/publicProfile.matchers.js:17]: there this could be a more specific and its own matcher

File: test/publicProfile.matchers.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+definePattern(
+  'aPublicProfile',
+  {
+    avatar: _.isAssetUrl,
+    banner: _.isUndefinedOr(_.isAssetUrl),
+    bio: _.isUndefinedOr(_.isString),
+    city: _.isUndefinedOr(_.isString),
+    country: _.isUndefinedOr(_.isString),
+    did: _.isDID,
+    displayName: _.isString,
```
Comment (deadlyicon) [inline test/publicProfile.matchers.js:14]: we also have some rules about this text that could be tested here

**Review by deadlyicon — CHANGES_REQUESTED** [review]

File: test/organizationMembership.matchers.spec.js (hunk 1 above)
Comment (nodatall) [reply test/organizationMembership.matchers.spec.js]: updated

File: test/organizationMemberships.matchers.js (hunk 2 above)
Comment (nodatall) [reply test/organizationMemberships.matchers.js]: cool, done

File: test/patterns.js (hunk 4 above)
Comment (nodatall) [reply test/patterns.js]: remoived

File: test/patterns.js (hunk 3 above)
Comment (nodatall) [reply test/patterns.js:44]: got it, removed

File: test/publicProfile.matchers.js (hunk 7 above)
Comment (nodatall) [reply test/publicProfile.matchers.js]: removed this whole matcher since I didn't sue it

File: test/publicProfile.matchers.js (hunk 6 above)
Comment (nodatall) [reply test/publicProfile.matchers.js]: see above

File: test/organizationMemberships.matchers.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    expect(target.memberUserDid).to.equal(memberUserDid);
+    expect(target.organizationApikey).to.equal(organizationApikey);
+  }
+);
+
+definePattern(
+  'anOrganizationAdminMembershipBetween',
+  (target, { organizationApikey, memberUserDid }) => {
+    expect(target).to.be.anOrganizationAdminMembership;
+    expect(target).to.be.anOrganizationMembershipBetween;({
```
Comment (deadlyicon) [inline test/organizationMemberships.matchers.js:48]: `;` ????
```suggestion
    expect(target).to.be.anOrganizationMembershipBetween;({
```

**Review by deadlyicon — CHANGES_REQUESTED** [review]
fix you matcher specs because they failed to throw in cases that are bugged by these extra `;`s

File: test/organizationMemberships.matchers.js (hunk 8 above)
Comment (nodatall) [reply test/organizationMemberships.matchers.js]: whoops, fixed

File: test/organizationMembership.matchers.spec.js [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    }).to.be.anOrganizationMembership();
+  });
+
+  it('anOrganizationCuratorMembership', function() {
+    expect(validOrganizationMembership).to.not.be.anOrganizationCuratorMembership();
+    expect({
+      ...validOrganizationMembership,
+      admin: undefined,
+      curator: true,
+    }).to.be.anOrganizationCuratorMembership();
```
Comment (deadlyicon) [inline test/organizationMembership.matchers.spec.js:32]: shouldn't each of these tests have something like
```js
expect(() => {
  expect(bad).to.be.aGood()
}).to.throw(`expected bad to be good`);
```

File: test/organizationMemberships.matchers.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  target => {
+    expect(target).to.be.anOrganizationMembership();
+    expect(target).to.matchPattern({ curator: true, '...': 1 });
+  }
+);
+
+definePattern(
+  'anOrganizationMembershipBetween',
+  (target, { organizationApikey, memberUserDid }) => {
+    expect(target).to.be.anOrganizationMembership;
```
Comment (deadlyicon) [inline test/organizationMemberships.matchers.js:38]: this is not being called

Comment (nodatall) [reply test/organizationMemberships.matchers.js]: https://media.giphy.com/media/3o6wrDdjMdUlO8PovS/giphy.gif

File: test/organizationMembership.matchers.spec.js (hunk 9 above)
Comment (nodatall) [reply test/organizationMembership.matchers.spec.js:39]: aaah yeah good point, added to all tests

**Review by deadlyicon — APPROVED** [review]

**Pull request merged**